quickwit/quickwit-storage/src/split.rs

```diff
@@ -59,11 +61,34 @@ async fn range_byte_stream_from_payloads(
         );
     }
 
-    let body = Body::wrap_stream(stream::iter(bytestreams).flatten());
-    let concat_stream = ByteStream::new(body.into());
+    let body = Body::wrap_stream(stream::iter(bytestreams).map(StreamAdaptor).flatten());
+    let concat_stream = ByteStream::new(SdkBody::from_body_0_4(body));
     Ok(concat_stream)
 }
 
+// With sdk 1.0, ByteStream no longer implement Stream, despite having analogous functions
+// this adaptor is just meant to make it implmeent Stream for places where we really need it
+#[pin_project]
+struct StreamAdaptor(#[pin] ByteStream);
+
+impl Stream for StreamAdaptor {
+    type Item = Result<Bytes, aws_smithy_types::byte_stream::error::Error>;
+
+    fn poll_next(
+        self: Pin<&mut Self>,
+        ctx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Option<Self::Item>> {
+        self.project().0.poll_next(ctx)
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let (min_u64, max_u64) = self.0.size_hint();
```

Comment:
Nitpick... I think lower_bound / upper_bound is slightly better naming than min/max.

```suggestion
        let (lower_bound_u64, upper_bound_u64_opt) = self.0.size_hint();
```